presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestAggregations.java

```diff
@@ -660,23 +660,23 @@ public void test15WayGroupBy()
     @Test
     public void testApproximateCountDistinct()
     {
-        assertQuery("SELECT approx_distinct(custkey) FROM orders", "SELECT 996");
-        assertQuery("SELECT approx_distinct(custkey, 0.023) FROM orders", "SELECT 996");
-        assertQuery("SELECT approx_distinct(CAST(custkey AS DOUBLE)) FROM orders", "SELECT 1031");
```

Comment:
In theory, from what i understand the deviation shouldn't be more than a `2.3%`. We had larger deviation before when casting to `DOUBLE`, now we have bigger deviation when casting to `VARCHAR`. Have you tried to investigate it? Is this a hashing problem? Or the HLL implementation problem?

Reply:
Based on my understanding, we don't guarantee that the error of a single observation from `approx_distinct` is bound. What we guarantee is the bounded "standard error", which is "the standard deviation of the (approximately normal) error distribution over all possible sets" ([presto doc](https://prestodb.io/docs/current/functions/aggregate.html#approx_distinct)). 

In other words, we guarantee that on "average", the error of `approx_distinct` is bounded. But for a single instance, it may not. 

And we have [test](https://github.com/prestodb/presto/blob/master/presto-main/src/test/java/com/facebook/presto/operator/aggregation/AbstractTestApproximateCountDistinct.java#L95,115) for this. It runs `approx_distinct` for 500 times and compute the standard error.